### ISSUE TITLE
Magnification of extracted tiles

### ISSUE THREAD

**Issue by gilShamai** [prov-gigapath/prov-gigapath#31]
It is unclear if the MPP of the extracted should be 0.5, similar to the way the models were trained. 
It is reasonable that if the model was trained on MPP=0.5, it should be applied on the same magnification. However, there were no details on this matter in the code description.

**Comment by HanwenXuTHU** [comment]
Good suggestion! We will add more instructions next.

**Comment by gilShamai** [comment]
Dear Hanwen,
I am heading a digital pathology team and we are about to finalize our last
experiments to be submitted to Nature Medicine.
We would like to try to finetune your gigapath model's weights using our
data as part of the experiments. i.e. - continue training the SSL model
using our data in a supervised manner.
Is there a way to do that using your code?
I posted this issue on Github as well:
"I am working with a dataset comprising 10,000 patients, each with an
associated target. I am interested in fine-tuning the SSL feature extractor
in a supervised manner to extract tile-level features that are more
specific to my task.
Could you kindly provide guidance on how to implement this using your code?
If this functionality is not currently supported, would it be possible for
you to make it available?
Thank you for your time and assistance."
Thanks a lot.
Best,
Gil

**Comment by HanwenXuTHU** [comment]
Hi Dr.Gil, yes I feel excited that you're interested in fine-tuning our model. Can I get more background of this task? Is it a slide-level prediction task or tile-level? Our current pipeline is (1) extract the features with a frozen tile-encoder (2) fine-tune the slide encoder to adapt to specific tasks. If our task here could accept this pipeline, then the answer is yes and I'm happy to help set up your experiments. Also, feel free to directly email me at xuhw@cs.washington.edu, I'm looking forward to your work!

**Comment by gilShamai** [comment]
Dear Hanwen,
I hope this message finds you well. I just wanted to follow up with you on
my previous email about the tile encoder.
Given our large cohort size and its distinctive features, we'd like to
finetune the tile extractor. Thus, I just wanted to clarify and suggest two
things that either of them would help us:
1) Could you share the code for training the tile encoder?
2) Or, if an existing framework was used, could you clarify how it was
implemented?
This would be very helpful for our current study and may significantly
accelerate our submission to Nature Medicine.

Thank you for your time,
Best,
Gil

**Comment by HanwenXuTHU** [comment]
Ah I see. Unfortunately I didn't have existing codes for end-to-end
finetuning because end-to-end finetuning will be computationally intensive.
But I do have the code for DINOv2 finetuning and that's only for tile-level
tasks. Does the following work: (1) Finetune tile encoder on your large
cohort by distributing the slide-level labels to tile level. (2) Freeze the
tile encoder from stage 1, and then finetune the slide encoder. If this
works, I can set up this pipeline for you. I'm a bit busy these days so
sorry for the late response. If this works, I will set it up for you before
Friday.

**Comment by gilShamai** [comment]
Dear Hanwen,
Thank you for the detailed response.
I am CC-ing Hen Davidov who is our deep learning expert and would be able
to follow up with more details on what we need.

Option (1) sounds like an appropriate solution for our problem. We weren't
aware that the DINOv2 finetuning code was still compatible with your tile
encoder. We'd very much appreciate help with setting that up, especially if
some modifications to the relevant open source code are necessary. Is that
the case?
Best,
Gil

**Comment by HanwenXuTHU** [comment]
Yes, I'm happy to help with setting up the option (1). Is your task the
classification task? If so, I can provide the finetuning script taking the
train.csv and test.csv as input. Each csv has the "image_path" and "labels"
as columns. You can then finetune by directly replacing the csv with your
customized csv files.

Best,

Hanwen

**Comment by gilShamai** [comment]
Dear Hanwen,
Thank you, that sounds exactly like what we need! We appreciate the help.
What kind of slide formats does your code accept?
CC-ing Hen again so you can communicate directly.
Best,
Gil

**Comment by gilShamai** [comment]
Dear Hanwen,

I wanted to follow up on my previous email in case it may have been missed.
Your assistance would be invaluable in helping us expedite our study, and I
would greatly appreciate any guidance or support you can provide.

Thank you very much for your time and consideration.

Best,
Gil

**Comment by HanwenXuTHU** [comment]
Yes it's ready. Can you provide me with your email address? Then I will
share it with you.

Regards,

Hanwen

**Comment by gilShamai** [comment]
sure,
***@***.***
OR
***@***.***

**Comment by gilShamai** [comment]
Dear Hanwen,
I haven't received an email from you yet.
Please let me know if you sent an email so that I can see it did not go to
the spam folder.
Best,
Gil

**Comment by gilShamai** [comment]
Dear Hanwen,
I know you must be busy so I am just pinging you again in case you've
missed my message.
Best
Gil
